pandas/tests/generic/test_generic.py

```diff
@@ -948,3 +948,8 @@ def test_deprecated_get_dtype_counts(self):
         df = DataFrame([1])
         with tm.assert_produces_warning(FutureWarning):
             df.get_dtype_counts()
+
+    @pytest.mark.parametrize("axis", [0, 1, "index", "columns", "rows"])
+    def test_interpolate_axis_argument(self, axis):
+        # GH 29142
+        assert DataFrame([0]).interpolate(axis=axis)
```

Comment:
Can you make a stronger test that asserts what the expected values are? We typically will do something like `result = df.interpolate(...)` and then `expected = ...` followed by `tm.assert_frame_equal(result, expected)`

Reply:
How is this? Since the data shouldn't change I compare it with its pre-interpolation self